Applications/SimpleShot/task_runner.cpp

```diff
@@ -110,25 +121,72 @@ const std::string getcwd_() {
 
 using LayerHandle = std::shared_ptr<ml::train::Layer>;
 
+/**
+ * @brief find model name by using strcmp
+ * @param backbone name
+ * @return int to identify the backbone
+ */
+
+int find_model_name(std::string backbone) {
+
+  const char *a = backbone.c_str();
+
+  if (std::strcmp(a, "conv4") == 0 || std::strcmp(a, "resnet50") == 0)
+    return 0;
+
+  if (std::strcmp(a, "facenet") == 0)
+    return 1;
+
+  return -1;
+}
+
+int new_classes;     /**< an integer value new classes to be added */
+int earlier_classes; /**< an integer value earlier classes which were present*/
+
 /**
  * @brief Create a Model with given backbone and varient setup
- *
+ * selected
  * @param backbone either conv4 or resnet50, hardcoded tflite path will be
  * selected
  * @param app_path designated app path to search the backbone file
  * @param variant "one of UN, L2N, CL2N"
+ * @param num_classes new_classes which are added
  * @return std::unique_ptr<ml::train::Model>
  */
-std::unique_ptr<ml::train::Model> createModel(const std::string &backbone,
-                                              const std::string &app_path,
-                                              const std::string &variant = "UN",
-                                              const int num_classes = 5) {
+
+std::unique_ptr<ml::train::Model>
+createModel(const std::string &backbone, const std::string &app_path,
+            const std::string &variant = "UN",
+            const int num_classes = new_classes) {
   auto model = ml::train::createModel(ml::train::ModelType::NEURAL_NET,
                                       {"batch_size=1", "epochs=1"});
 
-  LayerHandle backbone_layer = ml::train::layer::BackboneTFLite(
-    {"name=backbone", "model_path=" + getModelFilePath(backbone, app_path),
-     "input_shape=32:32:3", "trainable=false"});
+  /// Find backbone model name to be created
+
+  int model_name = find_model_name(backbone);
+  LayerHandle backbone_layer;
+
+  switch (model_name) {
+  case 0:
+    backbone_layer = ml::train::layer::BackboneTFLite(
+      {"name=backbone", "model_path=" + getModelFilePath(backbone, app_path),
+       "input_shape=32:32:3", "trainable=false"});
+    break;
+
+  case 1:
+    backbone_layer = ml::train::layer::BackboneTFLite(
+      {"name=backbone", "model_path=" + getModelFilePath(backbone, app_path),
+       "input_shape=112:112:3", "trainable=false"});
+
+    break;
+
+  default:
+    std::cout << "Wrong Model Name\n";
```

Comment:
I guess this should go to error handling routine instead of just breaking (eg) throw)

Reply:
Okay

Reply:
Please Check now